sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_client.py

```diff
@@ -165,6 +194,25 @@ def decrypt(self, algorithm, ciphertext, **kwargs):
             :dedent: 8
         """
         self._initialize(**kwargs)
+        iv = kwargs.pop("iv", None)
+        if iv and not ("CBC" in algorithm or "GCM" in algorithm):
```

Comment:
It may be time to centralize this in a function like `validate_arguments(operation, algorithm, **kwargs)`.